katello/hooks/pre/30-upgrade.rb

```diff
@@ -37,6 +37,17 @@ def migrate_foreman
   Kafo::Helpers.execute('foreman-rake db:migrate')
 end
 
+def postgresql_10_upgrade
+  return unless facts[:os][:release][:major] == '7'
```

Comment:
I think this should be a guard around `upgrade_step :postgresql_10_upgrade`. The difference is that otherwise on EL8 you'll have a file that says it did the postgresql 10 upgrade which it didn't really.